packages/eslint-plugin/tests/rules/no-unnecessary-condition.test.ts

```diff
@@ -1138,6 +1138,55 @@ isString('falafel');
       `,
       options: [{ checkTypePredicates: true }],
     },
+
+    {
+      code: `
+const items: number[] | null = [1, 2, 3];
```

Comment:
You'll want to do it like this, I think, here and elsewhere:
```suggestion
declare const items: number[] | null;
```
TS ignores the `| null` because it can see that it's not null ([playgroun](https://typescript-eslint.io/play/#ts=5.9.3&fileType=.ts&code=MYewdgzgLgBAllApgWwgLhmArsgRogJwG0BdGAH0ywBtqYBeGIgRgBoYAmdgZhIG4AUAhQRBAejEwAegH4hAMxgAKAIIECAQwCeAOjgQ1mrUuGoAlGZgBvATBgS7jp87uzbMUJBDVEO6iABzJQByUwh4cI0wGA11bWCzQQBfIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6AewFtLlFaAM1hMy%2BSpya0AhskKj0URNC7RI4MAF8QmoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false))